app/uk/gov/hmrc/gform/sharedmodel/formtemplate/destinations/Destination.scala

```diff
@@ -98,8 +98,16 @@ object Destination {
     instructionPdfFields: Option[InstructionPdfFields],
     convertSingleQuotes: Option[Boolean],
     payload: Option[String],
-    payloadType: TemplateType
-  ) extends Destination with DestinationWithCustomerId
+    payloadType: TemplateType,
+    roboticsAsAttachment: Option[Boolean]
+  ) extends Destination with DestinationWithCustomerId {
+    def roboticsFileName(fileNamePrefix: String, roboticsFileExtension: String): String =
+      if (roboticsAsAttachment.getOrElse(false)) {
+        s"$fileNamePrefix." + roboticsFileExtension
+      } else {
+        s"$fileNamePrefix-robotic." + roboticsFileExtension
+      }
```

Comment:
When the file is not using a `-robotic` and is viewed as attachment, it should  affect the count of attachment in metadata.xml I believe.

This should say 1 instead of 0 on the form which is not using any real attachments and which has `roboticsAsAttachment: true`:  https://github.com/hmrc/gform/blob/main/app/uk/gov/hmrc/gform/objectstore/MetadataXml.scala#L46